.github/actions/docker-build/action.yaml

```diff
@@ -35,7 +35,8 @@ runs:
       shell: bash
 
     - name: Cache
-      uses: actions/cache@v3
+      uses: actions/cache@v4
```

Comment:
```suggestion
      uses: actions/cache/save@v4
```
Would this make sense or do you need to somehow reuse cache from previous runs for this?

Reply:
Does it mean that the `build-main` workflow, which runs daily for health checks, should be built from scratch without using the cache?

Reply:
![image](https://github.com/autowarefoundation/autoware/assets/10751153/f5e8eb8a-09c8-4e9f-833b-ad9ecd45d49a)

1. This way first we try to restore.
1. For the first run, it won't be available.
1. It will generate the cache
1. Then save the cache  if build is successful.

This way save and restore is done explicitly in the correct places, what do you think?

Reply:
It's not effective because it is under `docker build`. That's why I use `buildkit-cache-dance` action to extract `--mount=type=cache` cache and save it.

Reply:
The cache extraction process by the `buildkit-cache-dance` action is executed during the `Post Build 'Autoware'` action. It is automatically executed as a cleanup process after the all workflow process has ended, so it cannot be manually written as a step in the workflow.

Reply:
Then, in this case, where does the cache save action happens in your initial proposal, when does it upload to the cache?

Reply:
We had a talk with @youtalk and here is the summary:
- caching happens in the Post Build step here https://github.com/autowarefoundation/autoware/actions/runs/9751034427/job/26911761719#step:11:47
- We have to use `buildkit-cache-dance` because` docker buildx bake` does not support volume mount
   - https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md?utm_source=pocket_reader#run---mounttypebind
   - More info: https://docs.docker.com/build/ci/github-actions/cache/#cache-mounts

Considering all these, and the tests that were conducted, I think current way is already the most optimal to my current understanding.

Thanks @youtalk !